### PR TITLE
feat: Introduce 3.x and 4.x style of openshift manifests.

### PR DESCRIPTION
**Pull request by iocanel**
This pull request introduces `quarkus.openshift.flavor` which supports values: `v3` and `v4` (default).
Based on the specified value:

# 3.x 
- We add the label `app` which is needed by the openshift console.
- We remove optional from secrets and configmaps.

### CODE REVIEW

**Comment by iocanel** [comment]
fyi: @gsmet: we discussed about it on zulip.

File: integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftV3Test.java
Comment (geoand) [inline integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftV3Test.java:24]: Can we have a corresponding v4 test?

Comment (iocanel) [reply integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftV3Test.java:24]: V4 is the default, it has not special configuration

Comment (geoand) [reply integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftV3Test.java:24]: Yeah I know. But do we have a test that asserts the same thing as this test but checks for the v4 conditions?

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java
Comment (gsmet) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java:25]:
```suggestion
     * The OpenShift flavor / version to use.
     * Older versions of OpenShift have minor differences in the labels and fields they support.
     * This option allows users to have their manifests automatically aligned to the OpenShift 'flavor' they use.
```

**Review by gsmet — DISMISSED** [review]
Spotted some typos.

Agreed with @geoand that we need to be sure things are covered for v4. Maybe they are already but we need to check.

**Comment by gsmet** [comment]
@iocanel what's the status for the tests? Are we good?

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
@gastaldi I had no answer to my question above.

@iocanel what's the status for the tests? Can you confirm it's taken care of?

**Comment by gastaldi** [comment]
@gsmet I merged this once I saw tests for v3 and v4, hence why I dismissed your review

**Comment by gsmet** [comment]
Then either you or Ionnis should drop a message please. It's really hard to keep track of things if people just silently do things. Thanks.

**Comment by gastaldi** [comment]
Will do, but rest assured that I always check if all concerns are resolved before merging a PR (well most of the time :))

**Pull request merged**
